magma/lib/magma/model.rb

```diff
@@ -209,5 +222,20 @@ def json_document(attribute_names = nil)
           model.identity => identifier
       )
     end
+
+    def date_shift_root_record
```

Comment:
Hm...going to merge this PR in, but as I started working on the shifted_date_time attribute, I think this will morph and probably not be here.